### PR TITLE
Trustline Operation

### PR DESCRIPTION
**Pull request by Daniel-Wang**
### Features

- Now you can `add` and `remove` trust lines on Android!
- UI shows progress bar on loading
- Views are updated on completion

### Screenshot
trustline

### CODE REVIEW

File: app/src/main/java/blockeq/com/stellarwallet/services/networking/Horizon.kt [hunk 1]
```diff
@@ -144,6 +145,51 @@ class Horizon {
...
+        class ChangeTrust(private val listener: SuccessErrorCallback, private val asset: Asset,
+                          private val removeTrust: Boolean)
+            : AsyncTask<Void, Void, ErrorResponse>() {
+
+            override fun doInBackground(vararg params: Void?): ErrorResponse? {
+                Network.usePublicNetwork()
+
+                val server = Server(PROD_SERVER)
+                val sourceKeyPair = WalletApplication.session!!.keyPair
+                val limit = if (removeTrust) {
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/services/networking/Horizon.kt]: An alternative suggestion to keep it concise: `val limit = removeTrust ? "0.0000000" : Constants.MAX_ASSET_STRING_VALUE`

File: app/src/main/java/blockeq/com/stellarwallet/activities/AssetsActivity.kt [hunk 2]
```diff
@@ -117,4 +131,32 @@ class AssetsActivity : BasePopupActivity() {
...
+        progressBar.visibility = View.VISIBLE
+    }
+
+    override fun hideProgressBar() {
+        progressBar.visibility = View.GONE
+    }
+
+    override fun reloadDataForAdapter() {
+        if (NetworkUtils(this).isNetworkAvailable()) {
+
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/activities/AssetsActivity.kt]: Remove extra line spacing here and also below in this function

File: app/src/main/java/blockeq/com/stellarwallet/activities/AssetsActivity.kt [hunk 3]
```diff
@@ -44,16 +51,22 @@ class AssetsActivity : BasePopupActivity() {
     //region User Interface
 
     private fun bindAdapter() {
-        adapter = AssetsRecyclerViewAdapter(this, assetsList!!)
+        adapter = AssetsRecyclerViewAdapter(this, this, assetsList!!)
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/activities/AssetsActivity.kt]: this, this --> lol but I understand.

**Comment by MeghaBambra** [comment]
@Daniel-Wang super minor stuff. Looks good 👍

File: app/src/main/java/blockeq/com/stellarwallet/services/networking/Horizon.kt (hunk 1 above)
Comment (Daniel-Wang) [reply app/src/main/java/blockeq/com/stellarwallet/services/networking/Horizon.kt]: Yup I'll do that

File: app/src/main/java/blockeq/com/stellarwallet/activities/AssetsActivity.kt (hunk 3 above)
Comment (Daniel-Wang) [reply app/src/main/java/blockeq/com/stellarwallet/activities/AssetsActivity.kt]: 😂

**Pull request merged**